pkg/kubelet/dockershim/libdocker/client.go

```diff
@@ -71,11 +71,15 @@ type Interface interface {
 // Get a *dockerapi.Client, either using the endpoint passed in, or using
 // DOCKER_HOST, DOCKER_TLS_VERIFY, and DOCKER_CERT path per their spec
 func getDockerClient(dockerEndpoint string) (*dockerapi.Client, error) {
+	clientOpts := []dockerapi.Opt{
+		dockerapi.FromEnv,
+		dockerapi.WithAPIVersionNegotiation(),
+	}
 	if len(dockerEndpoint) > 0 {
 		klog.Infof("Connecting to docker on %s", dockerEndpoint)
-		return dockerapi.NewClient(dockerEndpoint, "", nil, nil)
+		clientOpts = append(clientOpts, dockerapi.WithHost(dockerEndpoint))
```

Comment:
the previous NewClient invocation:
1. didn't use FromEnv before
2. always called WithVersion("") which set `manualOverride = true` internally

Reply:
I'll drop the improvements and just keep this PR simpler :)